action/src/app.ts

```diff
@@ -60,51 +61,32 @@ async function run(stoatConfig: any) {
   const ghBranch = core.getInput('pr_branch_name');
   await waitForStoatDevServer(github.context.repo, ghBranch, repoSha);
 
-  core.info('Checking if prior steps succeeded...');
-  let stepsSucceeded = true;
+  core.info(`Fetching commit timestamp...`);
+  const ghCommitTimestamp = await getGhCommitTimestamp(octokit, github.context.repo, repoSha);
 
+  // find the current job
   const jobListResponse = await octokit.rest.actions.listJobsForWorkflowRun({
     owner: github.context.repo.owner,
     repo: github.context.repo.repo,
     run_id: github.context.runId
   });
-
-  // todo: in the future we may want to determine which job we're currently in
-  // with matrix jobs and such this can be difficult to determine
-  // see https://github.com/actions/toolkit/issues/550 and the other plethora of issues complaining about this
-  for (const job of jobListResponse.data.jobs) {
-    core.info(`Inspecting job "${job.name}"`);
-    for (const step of job.steps || []) {
-      core.info(`-- Step "${step.name}": ${step.conclusion}`);
-      if (step.conclusion !== null && step.conclusion !== 'skipped') {
-        stepsSucceeded = stepsSucceeded && step.conclusion === 'success';
-      }
-    }
-  }
-
-  core.info(`Prior steps succeeded: ${stepsSucceeded}`);
-
-  core.info(`Fetching commit timestamp...`);
-  const ghCommitTimestamp = await getGhCommitTimestamp(octokit, github.context.repo, repoSha);
-  // The context.job in @actions/github is GITHUB_JOB, which is the job id, not the name.
-  // It is different from the job name in the job list response. So we cannot use it to
-  // search for the job information. We use job run id instead.
-  // References:
-  // https://github.com/actions/toolkit/blob/main/packages/github/src/context.ts
-  // https://docs.github.com/en/actions/learn-github-actions/environment-variables
   const ghJobId = github.context.job;
-  const ghJobRunId = github.context.runId;
-  const ghJob: GithubJob | undefined = jobListResponse.data.jobs.find((j) => j.run_id === ghJobRunId);
-
+  const ghJob: GithubJob | undefined = jobListResponse.data.jobs.find((j) => j.name === ghJobId);
```

Comment:
Actually the latest change does not always work. My previous comments are correct. When `job` has a `name` field, the `j.name` here is no longer the job id, but the name. It is really silly that `github.context` does not expose more env variables.